api/app/references/api.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime
+
+from flask import g, request
+import flask_restful as restful
+from flask_restful import fields, marshal_with
+
+from app.events.models import Event
+from app import LOGGER
+from app.utils.errors import EVENT_NOT_FOUND, USER_NOT_FOUND
+
+from app.utils.auth import auth_optional, auth_required
+from app.utils.emailer import send_mail
+from app.references.repository import ReferenceRequestRepository as reference_request_repository
+from app.references.models import ReferenceRequest
+from app.references.mixins import ReferenceRequestsMixin, ReferenceRequestsFormMixin, ReferenceRequestsListMixin
+from app.utils import misc
+from app.users.repository import UserRepository as user_repository
+
+reference_request_fields = {
+    'id': fields.Integer,
+    'title': fields.String,
+    'firstname': fields.String,
+    'lastname': fields.String,
+    'relation': fields.String,
+    'email_sent': fields.DateTime,
+    'response_id': fields.Integer,
+    'email': fields.String
+}
+
+class ReferenceRequestAPI(ReferenceRequestsMixin, restful.Resource):
+
+    @marshal_with(reference_request_fields)
+    @auth_required
+    def get(self):
+        LOGGER.debug('Received get request for reference-request')
+        args = self.req_parser.parse_args()
+        return reference_request_repository.get_by_id(args['id']), 200
+
+    
+class ReferenceRequestListAPI(ReferenceRequestsListMixin, restful.Resource):
+
+    @marshal_with(reference_request_fields)
+    @auth_required
+    def get(self):
```

Comment:
Let's add some validation here to check that the response belongs to the authenticated user, otherwise we could have users querying for other people's reference request data.